### DISCUSSION TITLE
Kafka input: metadata field "kafka_timestamp_unix" only displays down to seconds

### DISCUSSION THREAD

**Discussion by Elfspool** [redpanda-data/connect#2248] [Q&A]
In Kafka input, there is a metadata field named kafka_timestamp_unix and you have a unix timestamp in it. My problem is that the precision of that value is only down to seconds, but i need it to down to milliseconds. Is it possible to get the unix timestamp in a message down to milliseconds?

**Comment by mihaitodor** [comment]
It will require an enhancement [here](https://github.com/benthosdev/benthos/blob/9cc3a162fe5dfd41bb8293002c2b607671fe6b45/internal/impl/kafka/input_sarama_kafka.go#L399) and [here](https://github.com/benthosdev/benthos/blob/9cc3a162fe5dfd41bb8293002c2b607671fe6b45/internal/impl/kafka/input_kafka_franz.go#L259), maybe adding another metadata field called `kafka_timestamp_unix_ms` computed via [`UnixMilli()`](https://pkg.go.dev/time#Time.UnixMilli). It will be a bit redundant, though... Alternatively, maybe adding an advanced bool config flag called `kafka_timestamp_meta_ms` that defaults to `false` would be a better idea.

**Reply by Elfspool** [reply]
A bool config flag that can alternate between seconds and milliseconds would be rally nice feature. My case is problematic because we get customer data that changes frequently and a customer could have two messages in the same second and therefore, we cannot see which message is the latest for a specific customer. Seconds is a eternity in computing ;)

Really fast answer, so thanks for that mihaitodor. This i one of the reasons we like benthos at Lindex. fast response in the community.

**Reply by mihaitodor** [reply]
PRs are welcome :)
